lib/gooddata/bricks/middleware/aws_middleware.rb

```diff
@@ -12,21 +12,48 @@ module Bricks
     class AWSMiddleware < Bricks::Middleware
       def call(params)
         params = params.to_hash
-        if params.key?('aws_client')
+        s3_config = get_s3_config(params)
+
+        unless s3_config.empty?
           GoodData.logger.info('Setting up AWS-S3 connection')
-          raise 'Unable to connect to AWS. Parameter "aws_client" seems to be empty' unless params['aws_client']
-          raise 'Unable to connect to AWS. Parameter "access_key_id" is missing' if params['aws_client']['access_key_id'].blank?
-          raise 'Unable to connect to AWS. Parameter "secret_access_key" is missing' if params['aws_client']['secret_access_key'].blank?
-          params['aws_client'] = rewrite_for_aws_sdk_v2(params['aws_client'])
-          symbolized_config = GoodData::Helpers.symbolize_keys(params['aws_client'])
-          s3 = Aws::S3::Resource.new(symbolized_config)
-          params['aws_client']['s3_client'] = s3
+          if params.key?('aws_client')
+            params['s3_client'] = {}
+          elsif params['input_source'] && params['input_source']['bucket'].blank?
+            if params['s3_client']['bucket'].present?
```

Comment:
why we don't move bucket into get_s3_config(...), similar to the other fields.

Reply:
The AWS connect step (Aws::S3::Resource.new(symbolized_config)) will failed if it contains 'bucket' in the parameters. Bucket shoud be specified when we download file.